src/extension.ts

```diff
@@ -1,176 +1,208 @@
+                return themeInfo.fold(acc, themes => {
+                  return acc.concat(themes.filter((theme) => choosedSide === ThemeTypes.Both || theme.uiTheme === choosedSide).map((theme) => theme.id || theme.label));
+                });
+              }
+            }
+            // else {
+            //   console.log('here');//vscode-git-ui
+            // }
+            return acc;
+          },
+          [] as Array<string>
+        );
+
+        try {
+          if (excludeRegex) {
+            result = result.filter(themeName => !excludeRegex.test(themeName));
+          }
+        } catch (err) {
+          vscode.window.showErrorMessage(err);
+        }
+        finally {
+          r(result);
+        }
+      }
+      else {
+        c(null);
+      }
+
+    });
+  });
 
-      return themeInfo.fold(acc, themes => {
-        return acc.concat(themes.map(getThemeName));
-      });
-    },
-    [] as Array<string>
-  );
+  return promise;
 }
 
 async function saveThemes(
   themes: string[],
-  messageType: SaveThemeMessage,
-  messageTheme?: string,
+  message: string,
   userSettings = getUserSettings()
 ): Promise<void> {
   await userSettings.update('randomThemeSwitcher.themeList', themes, true);
 
-  vscode.window.showInformationMessage(getSaveMessages(messageType, { number: themes.length, theme: messageTheme }));
+  vscode.window.showInformationMessage(message);
 }
 
-function getThemeList(
+async function getThemeList(
   extensionConfig: vscode.WorkspaceConfiguration,
   userSettings: vscode.WorkspaceConfiguration
-): string[] {
-  const themeList: string[] = extensionConfig.get('themeList', []);
+): Promise<string[]> {
+  return new Promise(async (r, c) => {
+    const themeList: string[] = extensionConfig.get('themeList', []);
 
-  if (themeList.length === 0) {
-    const installedThemes = getInstalledThemes();
-    saveThemes(installedThemes, 'copyall');
-    return installedThemes;
-  }
+    if (themeList.length === 0) {
+      const installedThemes = await getInstalledThemes();
+      saveThemes(installedThemes, Messages.CopiedTheme(installedThemes.length));
+      r(installedThemes);
+    }
 
-  const currentTheme = userSettings.get('workbench.colorTheme', '');
-  if (themeList.length === 1) {
-    vscode.window.showInformationMessage('Why only one theme ç_ç');
-    return themeList;
-  }
+    const currentTheme = userSettings.get('workbench.colorTheme', '');
+    if (themeList.length === 1) {
+      vscode.window.showInformationMessage('Why only one theme ç_ç');
+      r(themeList);
+    }
 
-  return themeList.filter(theme => theme !== currentTheme);
+    r(themeList.filter(theme => theme !== currentTheme));
+
+  });
 }
 
-function addCurrentTheme(extensionConfig = getExtensionConfig(), userSettings = getUserSettings()) {
-  const themeList = getThemeList(extensionConfig, userSettings);
+async function addCurrentTheme(extensionConfig = getExtensionConfig(), userSettings = getUserSettings()) {
```

Comment:
inserisci il tipo di ritorno della funzione, servirà in futuro per dei refactor

Reply:
Fatto !